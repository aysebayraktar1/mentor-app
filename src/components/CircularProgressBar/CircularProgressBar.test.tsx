import React from 'react'
import { render } from '@testing-library/react'
import { CircularProgressBar } from './index'

describe('CircularProgressBar', () => {
    test('renders progress bar with correct progress and label', () => {
        const { getByRole, getByTestId } = render(<CircularProgressBar progress={50} label="Test Progress Bar" />)

        const progressBar = getByRole('progressbar')
        expect(progressBar.getAttribute('aria-valuenow')).toBe('50')
        expect(progressBar.getAttribute('aria-label')).toBe('Test Progress Bar')

        const progressBarText = getByTestId('progress-bar-text')
        expect(progressBarText).toHaveTextContent('50%')
    })

    test('updates progress bar when progress prop changes', () => {
        const { getByTestId, rerender } = render(<CircularProgressBar progress={25} />)
        const progressBar = getByTestId('progress-bar-bar')

        expect(progressBar.getAttribute('stroke-dashoffset')).toBe('193.20794819577225')

        rerender(<CircularProgressBar progress={75} />)
        expect(progressBar.getAttribute('stroke-dashoffset')).toBe('64.40264939859077')
    })
    it('should match snapshot', () => {
        const { container } = render(<CircularProgressBar progress={50} />)
        expect(container.firstChild).toMatchSnapshot()
    })
})
