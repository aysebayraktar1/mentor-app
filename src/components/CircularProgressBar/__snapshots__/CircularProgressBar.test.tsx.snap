// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgressBar should match snapshot 1`] = `
<div
  class="sc-3782deb7-0 gTtqYG"
>
  <svg
    aria-label="Progress Bar"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    height="74"
    role="progressbar"
    version="1.1"
    viewBox="0 0 100 100"
    width="74"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      class="sc-3782deb7-1 poyps"
      cx="50"
      cy="50"
      r="41"
      stroke-width="9"
    />
    <circle
      class="sc-3782deb7-1 sc-3782deb7-2 poyps bplCss"
      cx="50"
      cy="50"
      data-testid="progress-bar-bar"
      r="41"
      stroke-dasharray="257.610597594363 257.610597594363"
      stroke-dashoffset="128.8052987971815"
      stroke-width="9"
    />
  </svg>
  <div
    class="sc-3782deb7-3 liFQok"
    data-testid="progress-bar-text"
  >
    50
    <span
      class="sc-3782deb7-4 bRVHYS"
    >
      %
    </span>
  </div>
</div>
`;
